river/command/input.zig

```diff
@@ -0,0 +1,238 @@
+// This file is part of river, a dynamic tiling wayland compositor.
+//
+// Copyright 2021 The River Developers
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const std = @import("std");
+const mem = std.mem;
+
+const server = &@import("../main.zig").server;
+const util = @import("../util.zig");
+const c = @import("../c.zig");
+
+const Error = @import("../command.zig").Error;
+const Seat = @import("../Seat.zig");
+const InputConfig = @import("../InputConfig.zig");
+const InputManager = @import("../InputManager.zig");
+
+pub fn listInputs(
+    allocator: *mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    var input_list = std.ArrayList(u8).init(allocator);
+
+    var it = server.input_manager.input_devices.first;
+    while (it) |node| : (it = node.next) {
+        try input_list.appendSlice(@tagName(node.data.device.type));
+        try input_list.appendSlice(": ");
+        try input_list.appendSlice(node.data.identifier);
+
+        for (server.input_manager.input_configs.items) |*input_config| {
+            if (mem.eql(u8, input_config.identifier, mem.sliceTo(node.data.identifier, 0))) {
+                try input_list.appendSlice(" (configured)");
+                break; // There will only ever be one InputConfig for any unique identifier;
+            }
+        }
+
+        try input_list.appendSlice("\n");
+    }
+
+    out.* = input_list.toOwnedSlice();
+}
+
+pub fn listInputConfigs(
+    allocator: *mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    var input_list = std.ArrayList(u8).init(allocator);
+
+    for (server.input_manager.input_configs.items) |*input_config| {
+        try input_list.appendSlice(input_config.identifier);
+        try input_list.appendSlice("\n");
+
+        if (input_config.event_state) |event_state| {
+            try input_list.appendSlice("\tevents: ");
+            try input_list.appendSlice(@tagName(event_state));
+            try input_list.appendSlice("\n");
```

Comment:
We can use the writer interface and std.fmt to make this code less verbose. 
```suggestion
            try writer.print("\tevents: {}\n", .{event_state});
```